### ISSUE TITLE
'assertion failed: !ptr.is_null()' on asmjs-unknown-emscripten

### ISSUE THREAD

**Issue by sunng87** [rust-lang/regex#307]
I was in progress porting handlebars to webassembly. And got this issue, only on asmjs and wasm32.

```
thread 'main' panicked at 'assertion failed: !ptr.is_null()', /buildslave/rust-buildbot/slave/nightly-dist-rustc-cross-rustbuild-linux/build/src/libpanic_unwind/emcc.rs:33
note: Run with `RUST_BACKTRACE=1` for a backtrace.

/home/nsun/tmp/hbs-test/target/asmjs-unknown-emscripten/debug/hbs-test.js:110
      throw ex;
      ^
5390400
```

Here is a minimum source file to reproduce this issue:
 
```rust
    let replace = Regex::new(">|<|\"|&").unwrap();
    let result = replace.replace_all("<html>", |cap: &Captures| {
        match cap.at(0) {
            Some("<") => "&lt;".to_owned(),
            Some(">") => "&gt;".to_owned(),
            Some("\"") => "&quot;".to_owned(),
            Some("&") => "&amp;".to_owned(),
            _ => "".to_owned(),
        }
    });
    println!("{}", result);
```

My rustc: rustc 1.15.0-nightly (71c06a56a 2016-12-18)

**Comment by BurntSushi** [comment]
Note that `thread_local 0.2.x` is known to not work on emscripten. It was fixed in `0.3.x`: https://github.com/Amanieu/thread_local-rs/commit/053d8128106e995e3fdd8b91fdad1ad02fa3417f

**Comment by sunng87** [comment]
I confirmed this is caused by thread_local 0.2.x. Update to 0.3.2 will fix this issue.
